src/service/resources-impl.js

```diff
@@ -387,15 +391,42 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build(this.forceBuild_);
+        this.schedulePass();
+      } else {
+        // Track open elements
+        this.openElements_.push(element);
+      }
+      this.buildReadyResources_();
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
+    const closedElements = [];
+    this.openElements_.forEach(el => {
+      if (this.documentReady_ || el.nextSibling) {
+        const resource = this.getResourceForElement(el);
+        resource.build(this.forceBuild_);
+        closedElements.push(el);
```

Comment:
Probably could also just store the indexes instead of the elements themselves to avoid checking the index in the cleanup loop.